src/org/traccar/protocol/ArnaviBinaryProtocolDecoder.java

```diff
@@ -0,0 +1,273 @@
+package org.traccar.protocol;
+
+import org.jboss.netty.buffer.ChannelBuffer;
+import org.jboss.netty.buffer.ChannelBuffers;
+import org.jboss.netty.channel.Channel;
+import org.traccar.BaseProtocolDecoder;
+import org.traccar.DeviceSession;
+import org.traccar.helper.Log;
+import org.traccar.model.Position;
+
+import java.net.SocketAddress;
+import java.nio.ByteBuffer;
+import java.nio.ByteOrder;
+import java.time.Instant;
+import java.util.Arrays;
+import java.util.Date;
+import java.util.LinkedList;
+import java.util.List;
+
+public class ArnaviBinaryProtocolDecoder extends BaseProtocolDecoder {
+
+    public ArnaviBinaryProtocolDecoder(ArnaviBinaryProtocol protocol) {
+        super(protocol);
+    }
+
+    private static final int PROTOCOL_VERSION1 = 0x22;
+    private static final int PROTOCOL_VERSION2 = 0x23;
+
+    private static final byte PACKET_DATA = 0x01;
+    private static final byte PACKET_TEXT = 0x03;
+    private static final byte PACKET_FILE = 0x04;
+    private static final byte PACKET_DATA_BINARY = 0x06;
+
+    private static final byte TAG_VOLTAGE_SUPPLY = 1;
+    private static final byte TAG_ID_KEY = 2;
+    private static final byte TAG_LATITUDE = 3;
+    private static final byte TAG_LONGITUDE = 4;
+    private static final byte TAG_COORD_PARAMS = 5;
+    private static final byte TAG_DEVICE_STATUS = 9;
+    private static final byte TAG_FREQ_0 = 20;
```

Comment:
I think it's better to just use ranges instead of having variable for each FREQ_X, VOLT_X and LLS_X.